azure/flow-runner-function/src/test/java/com/marklogic/dhf/azure/HttpResponseMessageMock.java

```diff
@@ -0,0 +1,81 @@
+package com.marklogic.dhf.azure;
+
+import com.microsoft.azure.functions.*;
+
+import java.util.Map;
+import java.util.HashMap;
+
+/**
+ * The mock for HttpResponseMessage, can be used in unit tests to verify if the
+ * returned response by HTTP trigger function is correct or not.
+ */
+public class HttpResponseMessageMock implements HttpResponseMessage {
+    private int httpStatusCode;
+    private HttpStatusType httpStatus;
+    private Object body;
+    private Map<String, String> headers;
+
+    public HttpResponseMessageMock(HttpStatusType status, Map<String, String> headers, Object body) {
+        this.httpStatus = status;
+        this.httpStatusCode = status.value();
+        this.headers = headers;
+        this.body = body;
+    }
+
+    @Override
+    public HttpStatusType getStatus() {
+        return this.httpStatus;
+    }
+
+    @Override
+    public int getStatusCode() {
+        return httpStatusCode;
+    }
+
+    @Override
+    public String getHeader(String key) {
+        return this.headers.get(key);
+    }
+
+    @Override
+    public Object getBody() {
+        return this.body;
+    }
+
+    public static class HttpResponseMessageBuilderMock implements HttpResponseMessage.Builder {
+        private Object body;
+        private int httpStatusCode;
+        private Map<String, String> headers = new HashMap<>();
+        private HttpStatusType httpStatus;
+
+        public Builder status(HttpStatus status) {
+            this.httpStatusCode = status.value();
+            this.httpStatus = status;
+            return this;
+        }
+
+        @Override
+        public Builder status(HttpStatusType httpStatusType) {
+            this.httpStatusCode = httpStatusType.value();
+            this.httpStatus = httpStatusType;
+            return this;
+        }
+
+        @Override
+        public HttpResponseMessage.Builder header(String key, String value) {
```

Comment:
for consistency, just use Builder ?

Reply:
HttpResponseMessageMock is a sample provided by Azure. Didn't want to make changes to it, unless its absolutely necessary. There might be other use cases (later on) that may require other variations.

Reply:
Same thing here, this is mostly auto-generated by the Azure Maven archetype. We can change it later once we actually try to run a flow.